test/files/books_upsert.csv

```diff
@@ -1,3 +1,4 @@
 id,title,magic_number,_fivetran_deleted,_fivetran_synced
 3,"The Hobbit",14,false,"2024-01-09T04:10:19.156057706Z"
-2,"The Two Towers",1,false,"2024-01-09T04:10:19.156057706Z"
\ No newline at end of file
+2,"The Two Towers",1,false,"2024-01-09T04:10:19.156057706Z"
+99,"null","magic-nullvalue",false,"2024-01-09T04:10:19.156057706Z"
```

Comment:
I'm a bit confused about why the `magic_number` is now suddenly a string. Is `"magic-nullvalue"` an Arrow-specific thing?

Reply:
No, it's a Fivetran thing... They send two special strings that can show up in a column of any type -- one special string 
 indicates a null value (`null_string`) and the other is only for update files and indicates that a value should not be updated (`null_string`). The values of these two magic strings are unique for each `WriteBatchRequest`.
That's why I am reading upsert and update files without types -- just as varchar columns -- because string values can occur.

Reply:
Gotcha, thanks for clarifying